lib/api/accounts/user-accounts.js

```diff
@@ -0,0 +1,143 @@
+'use strict'
+
+const express = require('express')
+const bodyParser = require('body-parser').urlencoded({ extended: false })
+const debug = require('../../debug').accounts
+
+const CreateAccountRequest = require('../../requests/create-account-request')
+const AddCertificateRequest = require('../../requests/add-cert-request')
+
+/**
+ * Returns an Express middleware handler for checking if a particular account
+ * exists (used by Signup apps).
+ *
+ * @param accountManager {AccountManager}
+ *
+ * @return {Function}
+ */
+function checkAccountExists (accountManager) {
+  return (req, res, next) => {
+    let accountUri = req.hostname
+    accountManager.accountUriExists(accountUri)
+      .then(found => {
+        if (!found) {
+          debug(`Account ${accountUri} is available (for ${req.originalUrl})`)
+          return res.sendStatus(404)
+        }
+        debug(`Account ${accountUri} is not available (for ${req.originalUrl})`)
+        next()
+      })
+  }
+}
+
+/**
+ * Returns an Express middleware handler for creating a new user account
+ * (POST /api/accounts/new).
+ *
+ * @param accountManager {AccountManager}
+ *
+ * @return {Function}
+ */
+function createAccount (accountManager) {
+  return (req, res, next) => {
+    let request
+
+    try {
+      request = CreateAccountRequest.fromParams(req, res, accountManager)
+    } catch (err) {
+      err.status = err.status || 400
+      return next(err)
+    }
+
+    return request.createAccount()
+      .catch(err => {
+        err.status = err.status || 400
+        next(err)
+      })
```

Comment:
> when it is built as a collection of functions which operate on a small set of data rather than many classes each with their own ad-hoc interface.

I have tried, as much as possible, to break up the logic of these requests into methods that each operate on a small set of data. It's very likely that I may have missed places, though; feel free to let me know if any large or unwieldy pieces jump out.